sql/src/main/java/io/crate/execution/dsl/projection/UpdateProjection.java

```diff
@@ -143,16 +179,18 @@ public int hashCode() {
         result = 31 * result + (requiredVersion != null ? requiredVersion.hashCode() : 0);
         result = 31 * result + uidSymbol.hashCode();
         result = 31 * result + Arrays.hashCode(returnValues);
+        result = 31 * result + Arrays.hashCode(outputs);
 
         return result;
     }
 
-
     @Override
     public void writeTo(StreamOutput out) throws IOException {
-        super.writeTo(out);
+        Symbols.toStream(uidSymbol, out);
         if (out.getVersion().onOrAfter(Version.V_4_1_0)) {
             out.writeBoolean(allOn4_1);
+        } else {
+            allOn4_1 = false;
```

Comment:
Not really happy to modify a property in a `writeTo` method but i think it makes sense because this states clearly that not all nodes are on allOn4_1.